python/pylibcudf/pylibcudf/types.pyx

```diff
@@ -69,6 +70,38 @@ cdef class DataType:
         """Get the scale associated with this data type."""
         return self.c_obj.scale()
 
+    def python_typecode(self) -> str:
```

Comment:
1. I'd suggest making these `functools.cached_property`
2. It would be nice to `raise NotImplementedError` for types that don't have a corresponding string

Reply:
Also for `python_typecode` maybe we should make this a "private" (`_python_typecode`), as this is very specific to `array.array` so far. What do you think?

Reply:
Thanks, will do

Reply:
Oh wait, does cython support cached properties of cdef classes?

Reply:
Yeah the class has no `__dict__` to do the lookup from.

Reply:
Ah gotcha. Looks like we actually have an issue about this https://github.com/rapidsai/cudf/issues/16748

Reply:
Nice!